data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/server/PeerForwarderHttpServerProvider.java

```diff
@@ -80,7 +80,8 @@ public Server get() {
 
 
         sb.maxNumConnections(peerForwarderConfiguration.getMaxConnectionCount());
-        sb.requestTimeout(Duration.ofMillis(peerForwarderConfiguration.getRequestTimeout()));
+        // Add 1 second to the request timeout to allow for processing other than the buffer writes
+        sb.requestTimeout(Duration.ofMillis(peerForwarderConfiguration.getRequestTimeout() + 1000));
```

Comment:
This looks like another instance of the problem I described in #975. I tend to lean towards making the request_timeout the actual timeout and the buffer write a small value thereof.